androidtv/constants.py

```diff
@@ -28,20 +30,29 @@
 #: Determine whether the device is awake
 CMD_AWAKE = "dumpsys power | grep mWakefulness | grep -q Awake"
 
+#: Define current app
+CMD_DEFINE_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && " + VAR_CURRENT_APP_EXTRACTION
+
 #: Get the current app
-CMD_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep mCurrentFocus) && CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP"
+CMD_CURRENT_APP = CMD_DEFINE_CURRENT_APP + ' && echo $CURRENT_APP'
+
+#: Define current app for a Google TV device
+CMD_DEFINE_CURRENT_APP_GOOGLE_TV = 'CURRENT_APP=$(dumpsys activity a . | grep mResumedActivity) && ' + VAR_CURRENT_APP_EXTRACTION
 
 #: Get the current app for a Google TV device
-CMD_CURRENT_APP_GOOGLE_TV = "CURRENT_APP=$(dumpsys activity a . | grep -E 'mResumedActivity' | cut -d ' ' -f 8) && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP"
```

Comment:
It looks like you changed the command here (old first, new second):

```
CURRENT_APP=$(dumpsys activity a . | grep -E 'mResumedActivity' | cut -d ' ' -f 8) ...
CURRENT_APP=$(dumpsys activity a . | grep mResumedActivity) ...
```

Why the change?  I don't own a Google TV device, so I can't test whether this works.

Reply:
For this particular change I looked at this [Home Assistant Community topic](https://community.home-assistant.io/t/new-chromecast-w-android-tv-integration-only-showing-as-off-or-idle/234424/30) (reference from [#207](https://github.com/JeffLIrion/python-androidtv/issues/207#issuecomment-721840720)). Indicating that both `mCurrentFocus` and `mFocusedApp` are no longer specified in the `dumpsys window windows` output.

Then I checked the alternative command (`dumpsys activity a . | grep mResumedActivity`) to obtain application activity details, which is also working on my Sony TV running Android 8.0.0, and noticed it has the same **ActivityRecord** entry as `mFocusedApp`.

```
mResumedActivity: ActivityRecord{35dbb2a u0 com.google.android.tvlauncher/.MainActivity t2719}
```
```
mFocusedApp=AppWindowToken{c791eb8 token=Token{a6bad1b ActivityRecord{35dbb2a u0 com.google.android.tvlauncher/.MainActivity t2719}}}
```

Rather than using the existing solution which relies on space (indentation) to extract the application activity details, I decided to change this approach to be more generic by looking for the **ActivityRecord** instead.